python/simulation.py

```diff
@@ -4279,6 +4279,23 @@ def change_k_point(self, k):
                         py_v3_to_vec(self.dimensions, self.k_point, self.is_cylindrical)
                     )
 
+    def change_m_number(self, m):
+        """
+        Change the `m` number (the angular ϕ dependence).
+        """
+        self.m = m
+
+        if self.fields:
+            needs_complex_fields = not (not self.m or self.m == 0)
+
+            if needs_complex_fields and self.fields.is_real:
+                self.fields = None
+                self._is_initialized = False
+                self.init_sim()
```

Comment:
It would be better to just add a boolean argument to `fields::use_real_fields` so that we can switch back to real fields without initializing.  And then this could be called by `fields::change_m_number` for `m ≠ 1`